plugins/module_utils/network/sonic/config/bgp_af/bgp_af.py

```diff
@@ -245,6 +246,30 @@ def get_modify_advertise_request(self, vrf_name, conf_afi, conf_safi, conf_addr_
 
         return request
 
+    def get_modify_route_advertise_list_request(self, vrf_name, conf_afi, conf_safi, conf_addr_fam):
+        request = []
+        route_advertise = []
+        afi_safi = ('%s_%s' % (conf_afi, conf_safi)).upper()
+        route_advertise_list = conf_addr_fam.get('route_advertise_list', [])
+        if route_advertise_list:
+            for rt_adv in route_advertise_list:
+                advertise_afi_safi = rt_adv.get('advertise_afi_safi', None)
+                route_map = rt_adv.get('route_map', None)
+                if advertise_afi_safi:
+                    advertise_afi_safi = '%s_UNICAST' % advertise_afi_safi.upper()
```

Comment:
The usage of "afi_safi" here is confusing because it means two different things. It would be more clear (reducing the chance of coding errors if this is modified in the future) if there were two different names: advertise_afi and advertise_afi_safi. A corresponding change in the argspec to call the argument "advertise_afi" would help to make this consistent.

If this were done, the current line 256 would become:

```
advertise_afi = rt_adv.get('advertise_afi', None)
```

and the current line2 258 and 259 would become:

```
if advertise_afi:
    advertise_afi_safi = '%s_UNICAST' % advertise_afi.upper()
```